### DISCUSSION TITLE
Package "includes" functionality

### DISCUSSION THREAD

**Discussion by hughetop** [AcademySoftwareFoundation/rez#1405] [New Features]
Some 3rd party DCCs come with their own python interpreter. I've made custom python packages for them that are basically just:
```
name = 'python'
version = '3.10.2.blender'
variants = [['platform-windows', 'arch-AMD64']]
def commands():
    import sys
    if sys.platform == 'win32':
        # this needs to double up on quotes because of spaces in directories
        alias('python', r'"C:\Program Files\Blender Foundation\Blender 3.1\3.1\python\bin\python.exe"')
```
and the `blender` package has `requires = ['python-3.10.2.blender']`. This makes it so that anything that uses blender will also have a python package of the correct version, and it means I don't have to locally install python-3.10 on every computer. It's been working well, I can use this package to `rez-pip --install` things like PySide2 and it builds the correct variant.

Interestingly, blender's python interpreter comes with a few extras, like requests and numpy. So the new feature request is some kind of "includes" where this custom python package can say "requests and numpy are included in this package" so that I don't have to make new variants for requests and numpy only for them to be ignored at runtime. I was hoping ephemerals could do this, but maybe I just don't understand how to use them.

**Comment by instinct-vfx** [comment]
Unless i am missing something i think this would be covered by the "provides" feature. See here: https://github.com/AcademySoftwareFoundation/rez/issues/673

**Reply by hughetop** [reply]
Interesting! Yes that looks like exactly what I'm looking for! I believe in the meantime I can create "fake" variants for requests and numpy for python-3.10.2.blender that are empty.

Edit: fake variants seems to work as long as I add something like:
```
def commands:
    if 'blender' not in this.root:
        env.PYTHONPATH.append('{root}/python')
        ...
```

**Reply by herronelou** [reply]
I'm having similar issues with Nuke, keeping an eye on #1100, it's clashing all the way down with the `typing` module and it's not pretty :)

**Reply by JeanChristopheMorinPerso** [reply]
If I'm not mistaken, the `typing` module issue in Nuke can be fixed and it doesn't require the `provides` feature. You can probably ask in the `#wg-python3` AWS slack channel.

**Reply by herronelou** [reply]
I did some other workaround with the commands of the typing packages so that if nuke <13 is included it uses a shim that points back at Nuke's (well, shiboken's) typing module. Maybe `provides` wouldn't have solved it in this case because the built-in one is called `typing27`. Still, I have plenty of other cases, mostly around Qt, where `provides` would make my life a whole lot easier.
